### PR TITLE
rework docs for POSIX shared memory

### PR DESCRIPTION
**Pull request by tnguyen-rh**
- replace json w/ yaml
- add filename title and ‘source’ directive
- update cmd accordingly
- change ***\* blocks to ---- blocks
- markup /dev/shm
- use "provide" instead of "mount in"
- add intro blurb for example
- add callouts

@mrunalp @adellape 
Mostly normal changes; only potentially controversial item is the yaml.  WDYT?

### CODE REVIEW

File: dev_guide/shared_memory.adoc [hunk 1]
```diff
@@ -12,75 +12,55 @@ toc::[]
...
+<2> enables POSIX shared memory for `hello-container1` via `dshm`.
+<3> enables POSIX shared memory for `hello-container2` via `dshm`.
 ====
 
-****
-`$ oc create -f shared-memory.json`
-****
+----
+$ oc create -f shared-memory.yaml
+----
```
Comment (adellape) [inline dev_guide/shared_memory.adoc]: Even if it's fairly obvious, maybe add some brief context before this command, e.g.:

```
Create the pod using the *_shared-memory.yaml_* file:
```

File: dev_guide/shared_memory.adoc [hunk 2]
```diff
@@ -12,75 +12,55 @@ toc::[]
...
 There are two types of shared memory objects in Linux: System V and POSIX.
-The containers running in a pod share the IPC namespace of the pod infra container and so are able to share the System V shared memory objects. This document describes how they can also share POSIX shared memory objects.
+The containers in a pod share the IPC namespace of the pod infrastructure container and so are able to share the System V shared memory objects.
+This document describes how they can also share POSIX shared memory objects.
 
 
 == POSIX Shared Memory
-POSIX shared memory requires that a tmpfs be mounted at /dev/shm. The containers
- in a pod do not share their mount namespaces so we use volumes to mount in the same /dev/shm into each container in a pod.
+POSIX shared memory requires that a tmpfs be mounted at `/dev/shm`.
```
Comment (adellape) [inline dev_guide/shared_memory.adoc]: Maybe `*_/dev/shm_*` for the markup (filenames/directories)?

File: dev_guide/shared_memory.adoc [hunk 3]
```diff
@@ -12,75 +12,55 @@ toc::[]
...
+The containers in a pod do not share their mount namespaces so we use volumes to provide the same `/dev/shm` into each container in a pod.
```
Comment (adellape) [inline dev_guide/shared_memory.adoc]: Same re: `/dev/shm/`.

File: dev_guide/shared_memory.adoc (hunk 2 above)
Comment (tnguyen-rh) [inline dev_guide/shared_memory.adoc]: @adellape  Thanks for the tip.  I was looking around for example file/dir markup, but obviously didn't look hard enough.

**Comment by adellape** [comment]
+1 to YAML. Some comments inline.

**Comment by adellape** [comment]
LGTM after squash.

Will hold for @mrunalp tech review.

**Comment by mrunalp** [comment]
LGTM

**Comment by tnguyen-rh** [comment]
@adellape Squashed.

**Pull request merged**